### ISSUE TITLE
YourTinyTodo 2

### ISSUE THREAD

**Issue by alex-LE** [alex-LE/yourTinyTodo#90]
Hi guys, i've decided to start yourTinyTodo from scratch.
The current code is old and very hard to handle.

I plan to use:
- Laravel 4 as the PHP/Ajax Backend 
- Bootstrap 3 for layout

I currently work on the layout, please let me know what you think:

http://www.yourtinytodo.net/ytt2_preview

**Comment by Omegatcu** [comment]
Like the new layout.
Would like a Bundle for Symfony2 much more because everyone could implement that into any Project, but a stand alone App will also do fine.

**Comment by alex-LE** [comment]
What about using Laravel just for a simple REST API and build yourTinyTodo just as a JS only app? So maybe some one can create a REST API using Symfony or whatever.
I think about using a JS framework like AngularJS or Ember.js.
Using node.js would also be nice.

**Comment by Omegatcu** [comment]
Sounds good and would give every other Dev the ability to interact with MyTinyToDo regardless of the platform.
But if you make REST you can write a Class for it so you can add it in composer without the whole Laravel Framework.

**Comment by jniggemann** [comment]
Looks great!
Are you planning on making the user comments collapsible like in the old version?

**Comment by alex-LE** [comment]
@jniggemann a second collapsible area, so you can open the description and then the comments?

@all Yosko wrote a very interesting comment on the website and i want to share it with you:
_"Still I wonder: it looks more like and advanced task-list/project manager. Will it still be able to easily handle simple todo-list with only checkboxes when we want to?"_

What do you think?
I agree with him, due more and more features we moved away from the basic idea myTinyTodo had. It was build a simple todo list app, not as a big project management tool.
We might build a basic version of yTT and make it extensible with plugins?

**Comment by Omegatcu** [comment]
To be blunt:
I disagree very much with this statement.

A simple ToDo List would be a list with Checkboxes - over!

Category, Sorting, priority System - all that makes it a great todo app - yes also for Project management but also for simple Tasks you want to keep track off....

If you didn't need the priority system in MyTinyToDo you didn't have to use it. 
You didn't have to use the Comments or Tag System either - but it was convenient and useful.

I agree that not every feature should be put into the basic App, but I see no harm in implementing systems like Tags and Comments that are optional by nature.

If you feel thats too much then just make it optional by config or implement it in a way it can handle Plugins. This way you can extend the Project and get other Devs on board.

**Comment by alex-LE** [comment]
@Omegatcu you are right, features like comments or tags are important and should be part of the basic app. I had features like the timetable in mind, its not an important feature but might be useful for some people. 
Make yTT able to handle plugins seems to be very hard (when building a JS app). So the idea to make features optional seem so the best.

**Comment by Omegatcu** [comment]
@alex-LE Or you build the Main App in PHP again. Just add composer and only use the Packages you really need. You could use JS Templates and could support Plugins without being framework specific.

Of cource thats you choice you will have to make - how you want to build it and what tools to use.
But I feel it would be the way to go if you want the support of other dev´s and a reusable codebase.
No Point in doing all the work again in a couple of years...

**Comment by jniggemann** [comment]
@alex-LE: I meant collapsible just like in the old version where a task could be collapsed so only the "headline" is shown...
I like the feature set ytt currently has, if I were more proficient in programming jQuery or whatever you chose for the upcoming version, I'd drive it into a more GTD-like direction...
I imagine having tags like "@home", "@work", ... i.e. contexts, and entries show up in lists with the same name...

**Comment by alex-LE** [comment]
@jniggemann thats already working in the preview, just click on one of the headlines with the small arrow. The first task is open be default in the preview, just to show the description/comment stuff.

**Comment by jniggemann** [comment]
Ahh, didn't get that. Cool!

**Comment by hebbet** [comment]
are you going to release that preview soon?

so maybe other people can help you and provide pull request etc.

**Comment by alex-LE** [comment]
Just the preview itself? Yes i can release that, maybe in a extra repo.
The "subtask" part is still a bit buggy, so maybe i leave that for the beginning.
I'm still thinking about the technical background of the new version.

**Comment by alex-LE** [comment]
Its published as a separate branch

**Comment by hebbet** [comment]
Thanks. will send you some PRs :-)

**Comment by alex-LE** [comment]
You are welcome :)
I continued working on the angularjs part yesterday.

**Comment by hebbet** [comment]
hm tried to set it up, but all i get is 

Warning: require(/var/www/ytt/bootstrap/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/ytt/bootstrap/autoload.php on line 17

**Comment by hebbet** [comment]
ok, problem solved, pull requests are coming soon :-)
